controller/appcontroller.go

```diff
@@ -71,10 +74,12 @@ func NewApplicationController(
 	repoClientset reposerver.Clientset,
 	appResyncPeriod time.Duration,
 ) *ApplicationController {
-	db := db.NewDB(namespace, kubeClientset)
+	settingsMgr := settings_util.NewSettingsManager(kubeClientset, namespace)
+	db := db.NewDB(namespace, settingsMgr, kubeClientset)
 	kubectlCmd := kube.KubectlCmd{}
 	appStateManager := NewAppStateManager(db, applicationClientset, repoClientset, namespace, kubectlCmd)
 	ctrl := ApplicationController{
+		settingsMgr:           settingsMgr,
```

Comment:
This field is not used by the controller.

Also you need to update the role of the application controller to be able to get the argocd-cm configmap.

Reply:
fixed